### PR TITLE
Add `StreamMessage.mapError()`

### PR DESCRIPTION
**Pull request by ikhoon**
Motivation:

It would be useful to transform an error emitted by `StreamMessage`
into another.

Modifications:

- Add `StreamMessage.mapError()` that converts an error into another.
- Make `HttpRequest.mapError()` and `HttpResponse.mapError()` override
  `StreamMessage.mapError()`.
- Refactor `FuseableStreamMessage` to handle error functions.

Result:

- You can now transform an error of `StreamMessage` into another using
  a function.
  ```java
  StreamMessage streamMessage = StreamMessage.aborted(ClosedStreamException.get());
  StreamMessage transformed = streamMessage.mapError(ex -> {
      if (ex instanceof ClosedStreamException) {
          return new IllegalStateException(ex);
      } else {
          return ex;
      }
  });
  ```

### CODE REVIEW

**Comment by minwoox** [comment]
Could you check the failure? 😄

**Comment by ikhoon** [comment]
> Could you check the failure? 😄

Opps... 😅

**Review by minwoox — APPROVED** [review]
Thanks!

File: core/src/main/java/com/linecorp/armeria/common/HttpRequest.java [hunk 1]
```diff
@@ -557,4 +557,13 @@ default HttpRequest mapTrailers(Function<? super HttpHeaders, ? extends HttpHead
         });
         return of(headers(), stream);
     }
+
+    /**
+     * Transforms an error emitted by this {@link HttpRequest} by applying the specified {@link Function}.
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/common/HttpRequest.java:562]: How about adding a simple example to all three methods?

**Review by trustin — APPROVED** [review]

**Pull request merged**